kvstore/kvstore_server.cpp

```diff
@@ -1,18 +1,32 @@
 #include <kvstore_server.h>
 
 Status KvstoreServiceImpl::put(ServerContext* context, const PutRequest* request, PutReply* reply) {
-  // TODO: put
+  std::pair<std::string, std::string> pair(request->key(), request->value());
+  std::unique_lock<std::mutex> lock(mu_);
+  map_.insert(pair);
```

Comment:
your raw kvstore map should not be directly modified here. the kvstore data structure should not know anything about the servercontext, putrequests or putreplys - nothing chirp or grpc related. You need to create another class - KVStore - that has an unordered_map as it's member variable. It should also have put, get, remove but accepts only string values. Then this kvstore_server should instead have a member variable of KVStore and you should call instead call it something like kvstore.put(key, value)

https://docs.google.com/document/d/1TCxCbmyiO5mpiuvH8q9FZMXzt7nT2Drwecho8i3bW6g/edit#heading=h.tj35amwv5sbz

Reply:
changed